src/win/util.c

```diff
@@ -1387,3 +1390,180 @@ int uv__getpwuid_r(uv_passwd_t* pwd) {
 int uv_os_get_passwd(uv_passwd_t* pwd) {
   return uv__getpwuid_r(pwd);
 }
+
+
+int uv_os_getenv(const char* name, char* buffer, size_t* size) {
+  wchar_t var[MAX_ENV_VAR_LENGTH];
+  wchar_t* name_w;
+  DWORD bufsize;
+  size_t len;
+  int r;
+
+  if (name == NULL || buffer == NULL || size == NULL || *size == 0)
+    return UV_EINVAL;
+
+  /* Determine the size of the wide character name */
+  bufsize = MultiByteToWideChar(CP_UTF8, 0, name, -1, NULL, 0);
+
+  if (bufsize == 0)
+    return uv_translate_sys_error(GetLastError());
+
+  /* Convert the environment variable name to a wide character string */
+  name_w = uv__malloc(sizeof(wchar_t) * bufsize);
+
+  if (name_w == NULL)
+    return UV_ENOMEM;
+
+  bufsize = MultiByteToWideChar(CP_UTF8, 0, name, -1, name_w, bufsize);
+
+  if (bufsize == 0) {
+    uv__free(name_w);
+    return uv_translate_sys_error(GetLastError());
+  }
+
+  len = GetEnvironmentVariableW(name_w, var, MAX_ENV_VAR_LENGTH);
+  uv__free(name_w);
+  assert(len < MAX_ENV_VAR_LENGTH); /* len does not include the null */
+
+  if (len == 0) {
+    r = GetLastError();
+
+    if (r == ERROR_ENVVAR_NOT_FOUND)
+      return UV_ENOENT;
+
+    return uv_translate_sys_error(r);
+  }
+
+  /* Check how much space we need */
+  bufsize = WideCharToMultiByte(CP_UTF8, 0, var, -1, NULL, 0, NULL, NULL);
+
+  if (bufsize == 0) {
+    return uv_translate_sys_error(GetLastError());
+  } else if (bufsize > *size) {
+    *size = bufsize;
+    return UV_ENOBUFS;
+  }
+
+  /* Convert to UTF-8 */
+  bufsize = WideCharToMultiByte(CP_UTF8,
+                                0,
+                                var,
+                                -1,
+                                buffer,
+                                *size,
+                                NULL,
+                                NULL);
+
+  if (bufsize == 0)
+    return uv_translate_sys_error(GetLastError());
+
+  *size = bufsize - 1;
+  return 0;
+}
+
+
+int uv_os_setenv(const char* name, const char* value) {
+  wchar_t* name_w;
+  wchar_t* value_w;
+  DWORD bufsize;
+  int r;
+
+  if (name == NULL || value == NULL)
+    return UV_EINVAL;
+
+  name_w = NULL;
+  value_w = NULL;
+  r = 0;
+
+  /* Determine the size of the wide character name */
+  bufsize = MultiByteToWideChar(CP_UTF8, 0, name, -1, NULL, 0);
+
+  if (bufsize == 0) {
+    r = uv_translate_sys_error(GetLastError());
+    goto out;
+  }
+
+  /* Convert the environment variable name to a wide character string */
+  name_w = uv__malloc(sizeof(wchar_t) * bufsize);
+
+  if (name_w == NULL) {
+    r = UV_ENOMEM;
+    goto out;
+  }
+
+  bufsize = MultiByteToWideChar(CP_UTF8, 0, name, -1, name_w, bufsize);
+
```

Comment:
Couldn't this `MultiByteToWideChar` / `malloc` / `MultiByteToWideChar` operation be moved to a separate helper function as it is repeated multiple times?